tests/flakes.sh

```diff
@@ -845,3 +846,21 @@ nix store delete $(nix store add-path $badFlakeDir)
 
 [[ $(nix path-info      $(nix store add-path $flake1Dir)) =~ flake1 ]]
 [[ $(nix path-info path:$(nix store add-path $flake1Dir)) =~ simple ]]
+
+# Non-existant follows causes an error
+
+cat >$flakeFollowsA/flake.nix <<EOF
+{
+    description = "Flake A";
+    inputs.B = {
+        url = "path:./flakeB";
+        inputs.invalid.follows = "D";
+    };
+    inputs.D.url = "path:./flakeD";
+    outputs = { ... }: {};
+}
+EOF
+
+git -C $flakeFollowsA add flake.nix
+
+nix flake lock $flakeFollowsA 2>&1 | grep "error: B has a \`follows'-declaration for a non-existant input invalid!"
```

Comment:
If it's not too much work, do you think you could move that out of the `flakes` test (along with the other `follows` tests)? This big file gets really hard to understand and slow to run

Reply:
FWIW I moved all of the `flakeFollows*` test code into a separate `flake-follows.sh` file for #6621 but couldn't finish the changes because of a new bug, presumably from rebasing. I can push the incomplete state later today.

Reply:
Since @kha seems to have taken care of this already, I'd suggest to not do it here and to move it into `flakes-follows.sh` as soon as #6621 is ready if both of you are OK with that :)

Reply:
> FWIW I moved all of the flakeFollows* test code into a separate flake-follows.sh file for https://github.com/NixOS/nix/pull/6621
 
Uh [that's why](https://github.com/NixOS/nix/pull/6621#discussion_r913569695) I had this feeling of _dejà-vu_ when writing this comment. Sorry for the duplicate then, yes let's do the split in only one place, will be better :)